tests/foreman/longrun/test_leapp_upgrade.py

```diff
@@ -0,0 +1,55 @@
+"""Test for Upgrade using leapp
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: LEAPP
+
+:Assignee: shwsingh
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+
+def pre_upgrade_check(target_sat, target_version):
+    target_sat.execute(f'leapp preupgrade --target {target_version}')
+    target_sat.execute('yum install cockpit-leapp')
+
+
+def post_upgrade_cleanup_rhel7(target_sat, version):
+    # remove all packages from the exclude list
+    target_sat.execute('dnf config-manager --save --setopt exclude=' '')
+
+    # remove rhel7 packages including leapp packages
+    target_sat.execute('cd /lib/modules && ls -d *.el7*')
+
+    # remove weak modules from the old kernel
+    target_sat.execute(
+        f'[ -x /usr/sbin/weak-modules ] && /usr/sbin/weak-modules --remove-kernel {version}'
+    )
+
+    # Remove the old kernel from the bootloader entry.
+    target_sat.execute(f'/bin/kernel-install remove {version} /lib/modules/{version}/vmlinuz')
+
+    # Locate remaining RHEL7 packages
+    rhel7_packages = (
+        r"rpm -qa | grep -e '\.el[67]' | "
+        "grep -vE '^(gpg-pubkey|libmodulemd|katello-ca-consumer)' | sort''"
```

Comment:
more weird string formatting?
```suggestion
        "grep -vE '^(gpg-pubkey|libmodulemd|katello-ca-consumer)' | sort"
```